### PR TITLE
Car booking form

### PR DESCRIPTION
**Pull request by Pratishah**
ticket Link 
: https://abhirams.atlassian.net/browse/P3ECWFC-51
:https://abhirams.atlassian.net/browse/P3ECWFC-52

added booking form validation using form hook and ui changes acc to ticket  P3ECWFC-52
Screenshot (28)
Screenshot (29)

### CODE REVIEW

File: src/components/CarCard/CarCard.scss [hunk 1]
```diff
@@ -1,12 +1,38 @@
...
   p {
     display: flex;
     justify-content: space-between;
-    font-size: small;
-    font-weight: 600;
+    img,
+    svg {
+      color: $black;
+    }
+    font-size: 12px;
```
Comment (pant-manas) [inline src/components/CarCard/CarCard.scss]: @Pratishah there should be no value as px. please change that to rem or em

File: src/components/CarCard/CarCard.scss [hunk 2]
```diff
@@ -1,12 +1,38 @@
...
+  }
+}
+
+.card-container:hover {
+  text-decoration: none;
+  color: $black;
+  box-shadow: 1rem 1rem 1rem grey;
+  p {
+    font-weight: 500;
+    font-size: 12px;
```
Comment (pant-manas) [inline src/components/CarCard/CarCard.scss]: Same here for px. change it to rem and em

**Comment by Pratishah** [comment]
Screenshot (28)
Screenshot (29)

File: src/components/Navbar/TopNavbar.tsx [hunk 3]
```diff
@@ -1,40 +1,119 @@
...
 import "./TopNavbar.scss";
-import { Link } from "react-router-dom";
+import { Link, NavLink } from "react-router-dom";
 import Logo from "../../assets/logo.svg";
+import loggedProfile from "../../assets/man.png";
+import UnknownProfile from "../../assets/profile.svg";
+import { Button } from "react-bootstrap";
 
 function TopNavbar() {
+  const [isLoggedin, setisLoggedin] = useState(() => {
```
Comment (pant-manas) [inline src/components/Navbar/TopNavbar.tsx:10]: Camel casing. It should be setIsLoggedIn and isLoggedIn

File: src/components/Navbar/__test__/TopNavbar.test.tsx [hunk 4]
```diff
@@ -49,9 +61,20 @@ describe("Navbar", () => {
     expect(screen.getByText(/View All/i)).toBeInTheDocument();
   });
 
-  it("should render profile in the Navbar", () => {
+  it("should render profile icon if loggedin r", () => {
     render(<MockNavbar />);
-    const ProfileElement = screen.getByText(/Profile/i);
+    const isloggedin = localStorage.getItem("isloggedin");
+    const ProfileElement = screen.getByTestId("profileImage");
```
Comment (pant-manas) [inline src/components/Navbar/__test__/TopNavbar.test.tsx:67]: camelCasing of variable names

File: src/pages/Booking/Booking.scss [hunk 5]
```diff
@@ -1,42 +1,87 @@
...
     margin: 1rem;
     h6 {
-      font-weight: bold;
-      font-family: "Times New Roman", Times, serif;
-    }
-    h4 {
-      font-family: "Gill Sans", "Gill Sans MT", Calibri, "Trebuchet MS",
-        sans-serif;
+      font-weight: 800;
+      font-size: 32px;
```
Comment (pant-manas) [inline src/pages/Booking/Booking.scss:22]: Please make px into rem or em

File: src/pages/Booking/Booking.scss [hunk 6]
```diff
@@ -1,42 +1,87 @@
...
     }
+
     text {
-      font-size: small;
+      font-weight: lighter;
+      font-family: "'Roboto Condensed', sans-serif;";
+      font-size: 16px;
```
Comment (pant-manas) [inline src/pages/Booking/Booking.scss]: px to rem or em,

File: src/pages/Booking/Booking.scss [hunk 7]
```diff
@@ -1,42 +1,87 @@
...
       color: grey;
     }
     p {
-      font-size: smaller;
-      font-weight: bold;
+      margin-top: 0;
+      margin-bottom: 1rem;
+      font-weight: 800 !important;
+      font-family: "'Roboto Condensed', sans-serif;";
+      font-size: 20px;
```
Comment (pant-manas) [inline src/pages/Booking/Booking.scss]: px to rem or em

File: src/pages/Booking/Booking.tsx [hunk 8]
```diff
@@ -1,114 +1,181 @@
...
-    setValidated(true);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    mode: "onSubmit",
+  });
+  useEffect(() => {
+    document.title = "Xtreme Cars | Test Drive";
```
Comment (pant-manas) [inline src/pages/Booking/Booking.tsx]: Title is wrong

File: src/pages/Booking/Booking.tsx [hunk 9]
```diff
@@ -1,114 +1,181 @@
...
+  }, []);
+  const onSubmit = (formData: any) => {
+    console.warn("printing formdata from booking page", formData);
+    navigate("/booking_confirmation", { state: { car, formData } });
   };
 
-  const handleChange = (event: any) => {
-    const phone = event.target.value;
+  const { car } = useLocation().state as any;
+  console.log("printing car obj", car);
```
Comment (pant-manas) [inline src/pages/Booking/Booking.tsx]: Please remove console

File: src/components/Navbar/TopNavbar.tsx (hunk 3 above)
Comment (Pratishah) [reply src/components/Navbar/TopNavbar.tsx:10]: changing in topnav bar PR

**Review by pant-manas — COMMENTED** [review]
@kboodi looks got can you merge this

**Review by pant-manas — COMMENTED** [review]
@kboodi looks fine could you merge this

**Review by pant-manas — APPROVED** [review]

**Pull request merged**